pandas/core/strings.py

```diff
@@ -48,6 +48,8 @@ def cat_core(list_of_columns, sep):
     nd.array
         The concatenation of list_of_columns with sep
     """
+    if sep == '':
```

Comment:
can you show the benchmarks which change here?

Reply:
See the middle block of [this comment](https://github.com/pandas-dev/pandas/pull/22725#issuecomment-429097308)

Reply:
that was 8 months ago
pls show a new one

Reply:
This is a clear and unambiguous reduction in necessary computation, I fail to see how an ASV run is necessary (beyond what was done already, and especially since that code hasn't changed since then).

Maybe I'll get around to it in the next few weeks.

Reply:
I don't consider this totally unambiguous - is this only applicable to extremely wide data frames? Benchmarks and context would be very useful

Reply:
this adds complexity. unless this is like 2x faster would close this.

Reply:
> @WillAyd: I don't consider this totally unambiguous - is this only applicable to extremely wide data frames? Benchmarks and context would be very useful

Maybe the issue is not as apparent as I thought. Assuming a `list_of_columns=[s, t, u, v]` where (`s, t, u, v` are Series), then `cat_core` takes that list and interleaves `sep` to arrive at `[s, sep, t, sep, u, sep, v]` and pushes the whole thing into `np.sum`.

However, if the `sep=''`, there is effectively nothing to add, and we're just uselessly wasting cycles to interleave (and then add) an empty `sep`.

> @jreback: this adds complexity. unless this is like 2x faster would close this.

`.str.cat` has about 200LoC (not counting docstrings). Having 2 lines (1% of total) required to achieve 100% speedup sounds like an impossible standard for any perf-related change.

Based on the last runs in October (will try to do new ones as I said), I expect a 20-30% speedup for the (very common) case that `sep=''`. That should be more than enough to justify two trivial lines of code, IMO.

Reply:
> Based on the last runs in October (will try to do new ones as I said), I expect a 20-30% speedup for the (very common) case that sep=''. That should be more than enough to justify two trivial lines of code, IMO.

if that's true, then great. pls show benchmarks.

Reply:
Running the ASV is on my todo-list...